### ISSUE TITLE
Hue Sync does not find bridge

### ISSUE THREAD

**Issue by TobiS87** [marcelveldt/hass_emulated_hue#42]
Since the Entertainment  Zones word so much better now (can actually add all my desired lights to them) I tried syncing to Phillips HueSync for PC to get my "Ambilight" going. Unfotunately the program can not finde the bridge. I just searches the bridge forever. The prompt for link mode is triggered, but it never connects.

**Comment by Mr-Groch** [comment]
I think it may be the same problem as #41 . My Philips TV also doesn't discover Emulated Hue, but linking notification appear during discovery...

**Comment by marcelveldt** [comment]
version 0.0.8 is currently building to docker hub and should be available within an hour.
can you give that a test drive please ?

**Comment by TobiS87** [comment]
Thanks for the quick update. 0.0.8 really fixes some  things. First of all the bridge ist found and I see the screen where I am prompted to press the sync button. When I then go into link mode the pairing never happens. The timer just runs out without anything happening. 

The settings show the virtual bridge as found (and not paired) thorugh.

The Add-On log gives me the following messsage:

2021-01-01 16:21:13,587 WARNING emulated_hue.api -- Invalid/unknown request: <Request GET /auth/v1 >

**Comment by alexyao2015** [comment]
Thanks for the information! We'll need to test Hue Sync further to determine why it's not connecting.

**Comment by TobiS87** [comment]
I now dicovered that 0.0.8 crashes the hue app when trying to open the App. I wiped data on the app and now cant connect the emulated bridge in the hue app anymore. Ill do a completely fresh install (App an Add-On) later and try again.

As Hue Sync prompts that the bridge has to be connected over the hue app first maybe thats the issue... ill report back later.

**Comment by alexyao2015** [comment]
Thats strange. Maybe the certificate was changed to the the mac detection revisions?

Try uninstalling and reinstalling the add-on and the app.

@marcelveldt 
Are certificates overwritten automatically if the Mac is changed?

**Comment by TobiS87** [comment]
Completely fresh install worked - so everything back to normal in the hue app. 

Hue Snyc still doesnt work though. Same behavior as described an hour ago. Link mode is enabled but it will never connect. Patiently waiting for you guys doing your magic. ;)

**Comment by marcelveldt** [comment]
> Thats strange. Maybe the certificate was changed to the the mac detection revisions?
> 
> Try uninstalling and reinstalling the add-on and the app.
> 
> @marcelveldt
> Are certificates overwritten automatically if the Mac is changed?

No, but Hue apps are very restrictive about the certificate. So even if you automagically update the certificate on mac change, the Hue app will error out because the certificate doesn't match the mac. Maybe print an error in the logs when the certificate is (re)generated and store the mac in persistent config to detect this scenario.

In this case the issue is caused by the fact that the mac detection was fixed but that invalidated the certificate.

**Comment by alexyao2015** [comment]
Yes, we would likely still want to automatically update the certificate to avoid having to reinstall the add-on and losing the config. Reinstalling the app is a procedure that shouldn't hassle users that much.

**Comment by alexyao2015** [comment]
We can also read the Mac directly off the certificate to avoid having to store the Mac in the config. [See here.](https://github.com/diyhue/diyHue/blob/17552e0ca928b642c665fe43653e6fa48c00bff0/src/core/configManager/configStorage.py#L127)

**Comment by marcelveldt** [comment]
checked in a fix for the certificate/mac detection. certificate will now be recreated if the mac changes somehow.

**Comment by marcelveldt** [comment]
HUE Sync is currently not compatible with this addon. If you try to downgrade your version of Hue sync to a previous version it might still work though. Philips changed something that we need to figure out ;-)

**Comment by wessller** [comment]
Tried HueSync v 1.3.4.3 with 0.0.8. No luck. Waiting for 0.0.9 to be available.

**Comment by marcelveldt** [comment]
0.0.9 fixes the discovery issue so that *might* work with an older version of HUE Sync.

**Comment by wessller** [comment]
0.0.10. 
Hue Sync 1.3.4.3. Can connect to the Emulated Bridge, see the Entertainment areas, use the Scenes from Hue sync but no actual Syncing from either Video or Games.

Hue Sync 1.5.1.0 Can not connect to Emulated Bridge. Pairing requests is received

Hue Sync 1.5.3.3 Can not connect to Emulated Bridge. Pairing requests is received

**Comment by marcelveldt** [comment]
Good to know that you find the version from where it broke ;-)
Can you see a log stating that the entertainment mode server was started when you start Hue Sync ?

I'll try if I can lookup that version for MacOS too so I can have a test myself.

**Comment by Letzdeal3** [comment]
Hi all, error on Port 80 but nothing is running on port 80. Let me know if I am doing something wrong. Running latest Supervisor 2020.12.7 and OS 5.8

HUE Log:
2021-01-03 14:43:27,973 INFO  emulated_hue.config -- Auto detected listen IP address is 10.10.30.220
2021-01-03 14:43:27,989 INFO  hass-client -- Connecting to hassio/homeassistant
2021-01-03 14:43:27,991 ERROR emulated_hue.api -- Failed to create HTTP server at port 80: [Errno 98] error while attempting to bind on address ('0.0.0.0', 80): address already in use
2021-01-03 14:43:27,996 INFO  hass-client -- Connected to hassio/homeassistant
2021-01-03 14:43:28,017 INFO  emulated_hue.api -- Started HTTPS webserver on port 443

netstat:
~ $ netstat -l
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State
tcp        0      0 0.0.0.0:ssh             0.0.0.0:*               LISTEN
tcp        0      0 127.0.0.11:37889        0.0.0.0:*               LISTEN
tcp        0      0 0.0.0.0:8099            0.0.0.0:*               LISTEN
tcp        0      0 :::ssh                  :::*                    LISTEN
udp        0      0 127.0.0.11:36940        0.0.0.0:*
Active UNIX domain sockets (only servers)
Proto RefCnt Flags       Type       State         I-Node Path

Thanks 
Derek

**Comment by alexyao2015** [comment]
I'm seeing it find and attempt to pair with the bridge, but once registered, it does not progress further. 
Logs show GET `/api/nouser/config` and GET `/api/username/config`
Oddly enough, this would be indicative of incorrect data at either of these endpoints, but I've validated that the responses are exactly the same as an official bridge...

**Comment by alexyao2015** [comment]
Hue Sync 1.3.4.3 should work on the `fix/api-inconsistencies` for now. Still need to figure out why the new versions of Hue Sync fail to pair.

**Comment by WolfsWelpe** [comment]
I was just referred to this as my thread was closed. I don't use HueSync but Razer Synapse. It connects perfectly and also recognises the entertainment areas and the lights in them. Unfortunately, none of the lights respond even in the Hue app when I configure these areas (See thread Hue Entertainment - Lights not responding #69). They respond when arranging but no longer when testing the synchronisation.

**Comment by marcelveldt** [comment]
@WolfsWelpe the issue of non responding lights is being worked on, see #69 

As for Hue Sync, only older versions work atm as the HUE api got a major upgrade which we need to implement.
So Hue Sync 1.3.4.3 (or below) should work fine with current version of this addon.

**Comment by WolfsWelpe** [comment]
> @WolfsWelpe the issue of non responding lights is being worked on, see #69
> 
> As for Hue Sync, only older versions work atm as the HUE api got a major upgrade which we need to implement.
> So Hue Sync 1.3.4.3 (or below) should work fine with current version of this addon.

I'm still not using Hue Sync ^^ Just Razer Synapse but maybe there is the same problem with the API.

**Comment by BrunoN17** [comment]
Hi everyone!
I can confirm the same issue. The Android / iOS hue app works perfectly but on Windows it doesn't. Looks like it founds the bridge for half second and then it disappears. The discovery notification is triggered and nothing happens.

I can use some iOS third party apps like iLightShow but hueDynamic for Hue on Windows also doesn't work. After pairing it starts to say there was some communication issues with the bridge.

Let me know if there's anything I can do to help

Thanks!

**Comment by Trufax** [comment]
Is there any plans to update to the new API HueSync on WIndows does not work with the new version.

**Comment by M3te0r** [comment]
Hi, I'm facing the same issue with Hue Sync for Windows 10, the software does detect the emulated bridge and seems to be paired but it does not work further.

Tried with Hue Essentials but it just crashes at entertainment zone setup

It works well with the Hue module on Razez Synapse though
